iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1840,7 +1841,9 @@ private List<String> getPartitions(DDLOperationContext context, Configuration jo
       fetcher.initialize(job, HiveTableUtil.getSerializationProps());
       org.apache.hadoop.hive.ql.metadata.Table metaDataPartTable =
           context.getDb().getTable(hmstbl.getDbName(), hmstbl.getTableName(), "partitions", true);
+      SessionState.getSessionConf().set(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR, "partition,spec_id");
```

Comment:
https://github.com/apache/hive/blob/45867be6cb5308566e4cf16c7b4cf8081085b58c/ql/src/java/org/apache/hadoop/hive/ql/metadata/Table.java#L343-L344
`HiveIcebergSerDe `can only use the `session conf`, so here i set the `ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR` in the session conf, and then unset this propertity to avoid affecting other sql queries in the current seesion.